pkg/util/clusterdata/worker_profile_test.go

```diff
@@ -21,260 +21,213 @@ import (
 		},
 		{
-			name: "invalid cluster object",
-			client: func() machineclient.Interface {
-				return machinefake.NewSimpleClientset()
-			},
-			modifyOc: func(oc *api.OpenShiftCluster) {
-				oc.ID = "invalid"
-			},
+			name:   "invalid cluster object",
+			client: machinefake.NewSimpleClientset(),
 			wantOc: &api.OpenShiftCluster{
 				ID: "invalid",
 			},
+			givenOc: getGivenOc("invalid"),
 			wantErr: "parsing failed for invalid. Invalid resource Id format",
 		},
-	} {
-		t.Run(tt.name, func(t *testing.T) {
-			oc := &api.OpenShiftCluster{
-				ID: clusterID,
-			}
-
-			if tt.modifyOc != nil {
-				tt.modifyOc(oc)
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			//Given
+			clients := clients{
+				machine: tc.client,
 			}
 
 			e := machineClientEnricher{}
-			clients := clients{
-				machine: tt.client(),
-			}
 
-			e.SetDefaults(oc)
+			//When
+			e.SetDefaults(tc.givenOc)
+			err := e.Enrich(context.Background(), log, tc.givenOc, clients.k8s, clients.config, clients.machine, clients.operator)
 
-			err := e.Enrich(context.Background(), log, oc, clients.k8s, clients.config, clients.machine, clients.operator)
+			//Then
+			errorHandling.AssertErrorMessage(t, err, tc.wantErr)
 
-			if (err == nil && tt.wantErr != "") || (err != nil && err.Error() != tt.wantErr) {
-				t.Errorf("wanted err to be %s but got %s", err, tt.wantErr)
-			}
-			if !reflect.DeepEqual(oc, tt.wantOc) {
-				t.Error(cmp.Diff(oc, tt.wantOc))
+			if !reflect.DeepEqual(tc.givenOc, tc.wantOc) {
+
+				t.Error(cmp.Diff(tc.givenOc, tc.wantOc))
 			}
 		})
 	}
+
+}
+
+const (
+	mockSubscriptionID = "00000000-0000-0000-0000-000000000000"
+	mockVnetRG         = "fake-vnet-rg"
+	mockVnetName       = "fake-vnet"
+	mockSubnetName     = "cluster-worker"
+)
+
+var (
+	clusterID = fmt.Sprintf(
+		"/subscriptions/%s/resourceGroups/group/providers/Microsoft.RedHatOpenShift/openShiftClusters/cluster",
+		mockSubscriptionID,
+	)
+	workerSubnetID = fmt.Sprintf(
+		"/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s",
+		mockSubscriptionID, mockVnetRG, mockVnetName, mockSubnetName,
+	)
+)
+
+// This function creates a new MachineSet object with the given name and ProviderSpec.
+func createMachineSet(name string, ProvSpec machinev1beta1.ProviderSpec) *machinev1beta1.MachineSet {
+
+	// Create a new MachineSet object with the given name and namespace.
+	machset := &machinev1beta1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "openshift-machine-api",
+		},
+
+		// Specify the desired state for the MachineSet
+		Spec: machinev1beta1.MachineSetSpec{
+			Replicas: to.Int32Ptr(1),
+			Template: machinev1beta1.MachineTemplateSpec{
+				// Specify the desired configuration for the machine using ProviderSpec
+				Spec: machinev1beta1.MachineSpec{
+					ProviderSpec: ProvSpec,
+				},
+			},
+		},
+	}
+
+	// Return a pointer to the new MachineSet object.
+	return machset
```

Comment:
little nit, so we'd have consistency. 
also I am not sure the comments are really needed.  Also it doesn't create the object with the `given namespace`, as there is no given namespace in the function 
```suggestion
	// Create a new MachineSet object with the given name and namespace.
	return &machinev1beta1.MachineSet{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: "openshift-machine-api",
		},

		// Specify the desired state for the MachineSet
		Spec: machinev1beta1.MachineSetSpec{
			Replicas: to.Int32Ptr(1),
			Template: machinev1beta1.MachineTemplateSpec{
				// Specify the desired configuration for the machine using ProviderSpec
				Spec: machinev1beta1.MachineSpec{
					ProviderSpec: ProvSpec,
				},
			},
		},
	}
```

Reply:
made changes